### ISSUE TITLE
External Tools Configurations does not show/allow configuring previous runs

### ISSUE THREAD

**Issue by fabioz** [eclipse-platform/eclipse.platform.ui#1198]
Initially I thought that this was an issue due to my workspace, but it seems to happen even on a fresh workspace.

The image below should show the problem. I think this is a pretty critical issue as customizing external tools is hindered quite a bit by the bug...

![error_no_external_tools](https://github.com/eclipse-platform/eclipse.platform.ui/assets/117621/1c462443-67f8-42d4-9cca-bc3f54d61309)

**Comment by fabioz** [comment]
As a note, if I remove the `Filter Closed Projects` from the `External Tools Configurations` it does appear back.

So, maybe not as critical, but just very weird (as the ant build file is definitely in an open project as I had just made the launch).

**Comment by iloveeclipse** [comment]
Any errors in the log?

**Comment by fabioz** [comment]
> Any errors in the log?

No

**Comment by fabioz** [comment]
Actually, there's one (but I don't think is related):

```
java.io.FileNotFoundException: File D:\bin\eclipse_426_2022_12\jre\lib\ct.sym does not exist
	at org.eclipse.jdt.internal.compiler.util.CtSym.init(CtSym.java:128)
	at org.eclipse.jdt.internal.compiler.util.CtSym.<init>(CtSym.java:122)
	at org.eclipse.jdt.internal.compiler.util.JRTUtil.lambda$1(JRTUtil.java:203)
	at java.base/java.util.concurrent.ConcurrentHashMap.compute(Unknown Source)
	at org.eclipse.jdt.internal.compiler.util.JRTUtil.getCtSym(JRTUtil.java:200)
	at org.eclipse.jdt.internal.core.builder.ClasspathJrtWithReleaseOption.<init>(ClasspathJrtWithReleaseOption.java:77)
	at org.eclipse.jdt.internal.core.builder.ClasspathLocation.forJrtSystem(ClasspathLocation.java:176)
	at org.eclipse.pde.internal.build.site.PDEState.querySystemPackages(PDEState.java:675)
	at org.eclipse.pde.internal.build.site.PDEState.lambda$10(PDEState.java:555)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.base/java.util.stream.Streams$StreamBuilderImpl.tryAdvance(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline.findFirst(Unknown Source)
	at org.eclipse.pde.internal.build.site.PDEState.querySystemPackages(PDEState.java:556)
	at org.eclipse.pde.internal.build.site.PDEState.getSystemPackages(PDEState.java:493)
	at org.eclipse.pde.internal.core.TargetPlatformHelper.getSystemPackages(TargetPlatformHelper.java:391)
	at org.eclipse.pde.api.tools.internal.model.ApiBaseline.initialize(ApiBaseline.java:295)
	at org.eclipse.pde.api.tools.internal.model.ApiBaseline.initialize(ApiBaseline.java:243)
	at org.eclipse.pde.api.tools.internal.model.ApiBaseline.resolveSystemLibrary(ApiBaseline.java:473)
	at org.eclipse.pde.api.tools.internal.model.WorkspaceBaseline.addApiComponents(WorkspaceBaseline.java:98)
	at org.eclipse.pde.api.tools.internal.ApiBaselineManager.createWorkspaceBaseline(ApiBaselineManager.java:707)
	at org.eclipse.pde.api.tools.internal.ApiBaselineManager.getWorkspaceBaseline(ApiBaselineManager.java:627)
	at org.eclipse.pde.api.tools.internal.builder.ApiAnalysisBuilder.build(ApiAnalysisBuilder.java:377)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:1079)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:296)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:352)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:441)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:444)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:555)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:503)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:585)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:207)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:300)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

**Comment by iloveeclipse** [comment]
Could you please make sure D:\bin\eclipse_426_2022_12\jre is not used in any way by IDE? It looks like this is some old **JRE** but full **JDK** is required to most Java based tasks in Eclipse IDE.

**Comment by fabioz** [comment]
That is a `20.0.1` jre, but it should be used just to run Eclipse... The jdk is configured to point to the proper place:

![image](https://github.com/eclipse-platform/eclipse.platform.ui/assets/117621/5d2ce37f-7d98-4438-8d2a-14fd7db9564a)

**Comment by iloveeclipse** [comment]
> The jdk is configured to point to the proper place

But somewhere `D:\bin\eclipse_426_2022_12\jre` is still configured, see your stack trace above.

**Comment by fabioz** [comment]
I just removed that one from the installed jres (it seems it was there because it got it from the workplace that was started from a different version of eclipse -- the old was was 4.26, the current one has the jre in a different place -- I just removed it as it should not be used anyways).

I think I found the culprit on what happened though: I had a project (which was closed) which contained the same paths to the project I was working with.

My structure is something as:

```
/myprojectroot
/myprojectroot/plugins
/myprojectroot/plugins/org.myproject
```

and I had a closed project for `/myprojectroot` (as I just opened it when I wanted to edit some files in the root dir) and the open project for `/myprojectroot/plugins/org.myproject` -- for some reason the launch seemed to be bound to the closed `/myprojectroot` and not to the open project which I was interacting with when I created the configuration.
